cap/modules/deposit/serializers/schemas/json.py

```diff
@@ -109,12 +121,35 @@ def get_webhooks(self, obj):
     def get_deposit_schemas(self, obj):
         ui_schema = obj['deposit'].schema.deposit_options
         schema = current_jsonschemas.get_schema(
-            obj['deposit'].schema.deposit_path, with_refs=True, resolved=True)
+            obj['deposit'].schema.deposit_path, with_refs=True, resolved=True
+        )
+
+        # Adds readonly status for not allowed x-cap field
+        permission_info = copy.deepcopy(self.get_schema_permission_info(obj))
+        if permission_info:
+            schema = self.get_read_only_status(permission_info, schema)
+
         return dict(schema=copy.deepcopy(schema), uiSchema=ui_schema)
 
+    def get_read_only_status(self, permission_info, schema):
+        for x_cap_field in permission_info:
+            if self.can_user_edit_field(permission_info.get(x_cap_field)):
+                schema_field = schema.get("properties", schema)
```

Comment:
why start from `properties`? what if schema is of type `array` at the fist level?